db/pgp.js

```diff
@@ -0,0 +1,20 @@
+const config = require('../config');
+const { host, port, user, password, database } = config.db;
+
+// Loading and initializing the library:
+const pgp = require('pg-promise')({
+  // Initialization Options
+});
+
+const connectionStrings = {
+  test: `postgres://${user}:${password}@${host}:${port}/test`,
+  local: `postgres://${user}:${password}@${host}:${port}/${config.service.serviceName}`,
+  development: `postgres://${user}:${password}@${host}:${port}/${database}`,
+  production: `postgres://${user}:${password}@${host}:${port}/${database}`
+};
+
+// Creating a new database instance from the connection details:
+const db = pgp(connectionStrings[config.env]);
+
+// Exporting the database object for shared use:
+module.exports = db;
```

Comment:
is it worth supporting both knex and pgp? would that be more maintenance and overhead?

Reply:
It would be more effort to maintain both, but during testing of the process and performance it will be useful to keep both available

Reply:
fair enough.  You also explained you think pgp could be better for query optimisation so this is a sensible approach